### PR TITLE
Add support for query parameter when listing schedules

### PR DESCRIPTION
**Pull request by justinp-tt**
## What was changed

Add support for query parameter when listing schedules

## Checklist

1. Closes https://github.com/temporalio/sdk-go/issues/1539

2. How was this tested:

Added integration test cases

### CODE REVIEW

**Review by cretz — APPROVED** [review]
LGTM (though `Eventually` has a bit of a poor impl on the `testify` side, but probably fine here). Will leave open a bit to confirm @Quinn-With-Two-Ns has nothing to add.

Also, can you confirm you have privileges to "merge" this? If not, we can merge for you.

File: internal/internal_schedule_client.go [hunk 1]
```diff
@@ -199,6 +199,7 @@ func (sc *scheduleClient) List(ctx context.Context, options ScheduleListOptions)
 			Namespace:       sc.workflowClient.namespace,
 			MaximumPageSize: int32(options.PageSize),
 			NextPageToken:   nextToken,
+			Query:           options.Query,
```
Comment (Quinn-With-Two-Ns) [inline internal/internal_schedule_client.go:202]: Is there any issue on the server side when passing both a `Query` string and a `NextPageToken`? I don't think we do this anywhere else in the Go SDK and want to make sure the server does the correct thing.

Comment (dnr) [reply internal/internal_schedule_client.go:202]: ListWorkflowExecutions? The SDK doesn't call it but it exposes it to be used that way. IOW the token does not encapsulate the query, you still need the query.

Comment (Quinn-With-Two-Ns) [reply internal/internal_schedule_client.go:202]: yeah the SDK doesn't wrap `ListWorkflowExecutions ` so I wasn't sure on the contract.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Pull request merged**